miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/rest/HotRestController.java

```diff
@@ -47,4 +56,71 @@ public void setRows(List<List<String>> rows) {
     return MisoWebUtils.generateSpreadsheet(dto.getHeaders(), dto.getRows(), SpreadSheetFormat.valueOf(format), response);
   }
 
+  public static class ColumnDataDto {
+
+    private String heading;
+
+    private List<String> data = new ArrayList<>();
+
+    public ColumnDataDto(String heading) {
+      this.heading = heading;
+    }
+
+    public String getHeading() {
+      return heading;
+    }
+
+    public void setHeading(String heading) {
+      this.heading = heading;
+    }
+
+    public List<String> getData() {
+      return data;
+    }
+
+    public void setData(List<String> data) {
+      this.data = data;
+    }
+
+  }
+
+  @PostMapping("/import")
+  public @ResponseBody List<ColumnDataDto> importSpreadsheet(@RequestParam("file") MultipartFile file) throws IOException {
+    String extension = file.getOriginalFilename().substring(file.getOriginalFilename().lastIndexOf(".") + 1).toLowerCase();
+    SpreadsheetWrapper sheet = null;
+    if (extension.equals("xlsx")) {
```

Comment:
If you're going to go to the trouble of extracting and extension, I think it makes more sense to use a `switch` block. That, or don't bother and just do `filename.endsWith(".xlsx")`.